pkg/scale/decode.go

```diff
@@ -121,46 +119,94 @@ func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
 	switch in.(type) {
 	case *big.Int:
 		err = ds.decodeBigInt(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding big integer: %w", err)
+		}
 	case *Uint128:
 		err = ds.decodeUint128(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding Uint128: %w", err)
```

Comment:
nit I don't think we need to use capitalized words, since we don't need camelCase for error string messages.

```suggestion
				err = fmt.Errorf("decoding uint128: %w", err)
```